x/incentives/keeper/gov.go

```diff
@@ -0,0 +1,38 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+
+	"github.com/osmosis-labs/osmosis/v19/x/incentives/types"
+)
+
+func (k Keeper) HandleCreateGaugeProposal(ctx sdk.Context, p *types.CreateGroupsProposal) error {
+	for _, group := range p.CreateGroups {
+		incentivesModuleAddress := k.ak.GetModuleAddress(types.ModuleName)
+		// N.B: We force internal gauge creation here only because we don't have a straightforward
+		// way to escrow the funds from the prop creator to be used at time of prop execution (or returned if the prop fails).
+		// Once we have a way to do this, we can change the CreateGroups proto to allow for coins and numEpochsPaidOver and
+		// then modify it here as well.
+		_, err := k.CreateGroup(ctx, sdk.Coins{}, 0, incentivesModuleAddress, group.PoolIds)
```

Comment:
I think we should exprt this to types and reuse here in place of the mysterious 0:
https://github.com/osmosis-labs/osmosis/blob/5ab084a4ea13c9d38bacdb62b7228ab4c72dcfcc/x/incentives/keeper/gauge.go#L30

Reply:
Moved to types, exported, and replaced zero here and here

https://github.com/osmosis-labs/osmosis/pull/6547/commits/5e5fd9a0697f1d74676291a628cdb98e08a37c67
https://github.com/osmosis-labs/osmosis/pull/6547/commits/4435ae44e230d199dfb1afa7c2a184955398ca66